.github/workflows/ci.yaml

```diff
@@ -0,0 +1,141 @@
+
+  check_if_version_upgraded:
+    name: Check if version upgrade
+    # We run this only if it's a push on main or if it's a PR from a 
+    # branch (meaning not a PR from a fork). It would be more straightforward to test if secrets.NPM_TOKEN is 
+    # defined but GitHub Action don't allow it yet.
+    if: |
+      github.event_name == 'push' || 
+      github.event.pull_request.head.repo.owner.login == github.event.pull_request.base.repo.owner.login 
+    runs-on: ubuntu-latest
+    needs: 
+      - test_node
+      - test_deno
+    outputs:
+      from_version: ${{ steps.step1.outputs.from_version }}
+      to_version: ${{ steps.step1.outputs.to_version }}
+      is_upgraded_version: ${{ steps.step1.outputs.is_upgraded_version }}
+      is_pre_release: ${{steps.step1.outputs.is_pre_release }}
+    steps:
+    - uses: garronej/ts-ci@v2.0.2
+      id: step1
+      with: 
+        action_name: is_package_json_version_upgraded
+        branch: ${{ github.head_ref || github.ref }}
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: 
+      - check_if_version_upgraded
+    # We create a release only if the version have been upgraded and we are on a default branch
+    # PR on the default branch can release beta but not real release
+    if: |
+      needs.check_if_version_upgraded.outputs.is_upgraded_version == 'true' &&
+      (
+        github.event_name == 'push' ||
+        needs.check_if_version_upgraded.outputs.is_pre_release == 'true'
+      )
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.ref }}
+    - run: rm -r .github
+    - run: |
+        git branch -d latest || true
+        git push origin :latest || true
+    - run: |
+        git branch latest
+        git checkout latest
+    - uses: actions/setup-node@v3
+      with:
+        registry-url: https://registry.npmjs.org/
+    - uses: bahmutov/npm-install@v1
+    - run: yarn build
+    - run: npx -y -p denoify@1.2.1 enable_short_npm_import_path
+      env:
+        DRY_RUN: "0"
+    - run: |
+        if [ "$(npm show . version)" = "$VERSION" ]; then
+          echo "This version is already published"
+          exit 0
+        fi
+        if [ "$NODE_AUTH_TOKEN" = "" ]; then
+          echo "Can't publish on NPM, You need a NPM_TOKEN secret."
+          false
+        fi
+        EXTRA_ARGS=""
+        if [ "$IS_PRE_RELEASE" = "true" ]; then
+            EXTRA_ARGS="--tag next"
+        fi
+        npm publish $EXTRA_ARGS
+      env:
+        NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+        VERSION: ${{ needs.check_if_version_upgraded.outputs.to_version }}
+        IS_PRE_RELEASE: ${{ needs.check_if_version_upgraded.outputs.is_pre_release }}
+    - run: |
+        git config --global user.name "actions"
+        git config --global user.email actions@example.com
+        git add -A
+        git commit -am "Enabling shorter import paths"
+        git push origin latest
```

Comment:
I'm not sure I get this part, does it mean the action will create a "Enabling shorter import paths" commit every time there is a push on the main branch with a new version in the package.json?

Reply:
I removed the path shortener script because it's not useful for `ts-pattern`.  
The path shortener enables to import specific files like  
```typescript
import { isMatching } from "ts-pattern/is-matching";
```
instead of 
```typescript
import { isMatching } from "ts-pattern/dist/is-matching";
```  
It's useful when you want to enable users to cherry-pick what they import and for exposing extra things that have no business being in the default export, like for example a helper for a specific framework.  

I let you have a look at the updated workflow, I have added annotations.